lib/linter.js

```diff
@@ -230,17 +230,23 @@ export default class Linter {
     let fileConfig = this._moduleStatusCache.getConfigForFile(options);
 
     let ruleNames = new Set(fixableIssues.map((r) => r.rule));
+
     for (let ruleName of ruleNames) {
-      let rule = this._buildRule(ruleName, {
-        shouldFix: true,
-        filePath: options.filePath,
-        rawSource: currentSource,
-        fileConfig,
-      });
+      let templateInfos = extractTemplates(currentSource, options.filePath);
 
-      let visitor = await rule.getVisitor();
-      let { code } = transform(currentSource, () => visitor);
-      currentSource = code;
+      for (let templateInfo of templateInfos) {
+        let rule = this._buildRule(ruleName, {
+          shouldFix: true,
+          filePath: options.filePath,
+          rawSource: templateInfo.template,
+          fileConfig,
+        });
+
+        let visitor = await rule.getVisitor();
+        let { code } = transform(templateInfo.template, () => visitor);
+
+        currentSource = currentSource.replace(templateInfo.template, code);
```

Comment:
Can you add a test case for:

```js
// this one shouldn't be touched by a `--fix`
const bar = <template>{{! <button></button> }}</template>;
// this should get `--fix`'ed
const foo = <template><button></button></template>;
```

I think the current `string.replace` system will replace both `bar` and `foo` there. I believe that we need to manually splice the string contents, something like:

```js
let currentSource = `
const bar = <template>{{! <button></button> }}</template>;
const foo = <template><button></button></template>;`
// -> '\nconst bar = <template>{{! <button></button> }}</template>;\nconst foo = <template><button></button></template>;'

currentSource = currentSource.slice(0, 82) + `<button type="submit"></button>` + currentSource.slice(99)
// -> '\nconst bar = <template>{{! <button></button> }}</template>;\nconst foo = <template><button type="submit"></button></template>;'
```